src/main/java/Duke.java

```diff
@@ -1,10 +1,98 @@
+import java.util.Scanner;
+
 public class Duke {
+    private static boolean exitFlag;
+
     public static void main(String[] args) {
+        printIntroMsg();
+
+        // Initialize TaskHelper
+        TaskHelper taskHelper = TaskHelper.getInstance();
+
+        String userCommand;
+        Scanner in = new Scanner(System.in);
+
+        while(!exitFlag){
+            userCommand = in.nextLine();
+            final String[] split = userCommand.trim().split("\\s+", 2);
+            final String[] commandTypeAndParams = split.length == 2 ? split : new String[]{split[0], ""};
+            final String commandType = commandTypeAndParams[0];
+            final String commandArgs = commandTypeAndParams[1];
+
+            switch(commandType){
+            case "bye":
+                exitFlag = true;
+                break;
+            case "list":
+                listTasks();
+                break;
+            case "done":
+                if(!commandArgs.isEmpty()){
```

Comment:
Do add white spaces between 'if', the condition and the curly bracket; if statements should have the following form:
if (!commandArgs.isEmpty()) {

Reply:
Great catch. I missed out on that!